### DISCUSSION TITLE
border-opacity doesn't work in darkMode

### DISCUSSION THREAD

**Discussion by atharva-bhange** [tailwindlabs/tailwindcss#6164] [Help] [answered]
I have created a simple react + tailwind project with JIT and border-opacity doesn't seem to work in dark mode.
I recreated the issue in Tailwind Play here is the URL
https://play.tailwindcss.com/BclroAlp9e

I have seen this similar question in discussion https://github.com/tailwindlabs/tailwindcss/discussions/5757 . But the issue didn't describe anything about darkMode.

Should we create an issue for this ?

**Comment by NovusOpus** [comment, marked as answer]
There's no issue here, that's how it's suppose to work. You need to also add the `dark:` variant, in this case: `dark:border-opacity-50`.

Since `dark:` classes has a higher **Specificity** When using the **class strategy**`(darkMode :"class")` in your `tailwindconfig file` , in your case the generated css code from `dark:border-blue-700` is: 
```postcss
.dark .dark\:border-blue-700 {
  --tw-border-opacity: 1;
  border-color: rgba(29, 78, 216, var(--tw-border-opacity));
}
```
As you can see there's a `--tw-border-opacity: 1` property being generated as part of `border-blue-700` and resetting the opacity back to 1, or 100%.

So if you want the **Border Opacity of 50%**  to be the same in normal mode and dark mode, you have to add both: `border-opacity-50` and `dark:border-opacity-50`.

Source: [Tailwind Docs, There's even an example that's similar to your case](https://tailwindcss.com/docs/dark-mode#specificity-considerations)

**Reply by atharva-bhange** [reply]
Ok got it works.

As showed in example if i want border-opacity to go to 100 on focus i have to make a different focus for dark mode too like this
```focus:border-opacity-100 dark:focus:border-opacity-100 ```

Thanks for helping out!
